extensions/-SIPC-/time.js

```diff
@@ -70,20 +75,124 @@
               },
             },
           },
+          {
+            blockType: Scratch.BlockType.LABEL,
+            text: "Time Calculations",
+          },
+          {
+            opcode: "calculatetimedurationfromdate",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "time duration from [DATE] to current date in [TIME_MENU]",
+            arguments: {
+              DATE: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "2006-04-16",
+              },
+              TIME_MENU: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "Time",
+                defaultValue: "day",
+              },
+            },
+          },
+          {
+            opcode: "calculatetimedurationfromtime",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "time duration from [START_HOUR]:[START_MINUTE] to current time in [TIME_MENU]",
```

Comment:
you dont need two inputs for the time here

Reply:
you should also rename this and the above to be more like the below

"difference between 12:30 and current time in days"